src/app/files/fileManager.js

```diff
@@ -42,6 +52,202 @@ class FileManager extends Plugin {
+   */
+  async _handleIsDir (path, message) {
+    const isDir = await this.isDirectory(path)
+
+    if (!isDir) {
+      throw createError({ code: 'ENOTDIR', message })
+    }
+  }
+
+  /** The current opened file */
+  file () {
+    const file = this.currentFile()
+
+    if (!file) throw createError({ code: 'ENOENT', message: 'No file selected' })
+    return file
+  }
+
+  /**
+   * Verify if the path exists (directory or file)
+   * @param {string} path path of the directory or file
+   * @returns {boolean} true if the path exists
+   */
+  exists (path) {
+    const provider = this.fileProviderOf(path)
+    const result = provider.exists(path, (err, result) => {
+      if (err) return false
+      return result
+    })
+
+    return result
+  }
+
+  /**
+   * Verify if the path provided is a file
+   * @param {string} path path of the directory or file
+   * @returns {boolean} true if path is a file.
+   */
+  isFile (path) {
+    const provider = this.fileProviderOf(path)
+    const result = provider.isFile(path)
+
+    return result
+  }
+
+    /**
+   * Verify if the path provided is a directory
+   * @param {string} path path of the directory
+   * @returns {boolean} true if path is a directory.
+   */
+  isDirectory (path) {
+    const provider = this.fileProviderOf(path)
+    const result = provider.isDirectory(path)
+
+    return result
+  }
+
+  /**
+   * Open the content of the file in the context (eg: Editor)
+   * @param {string} path path of the file
+   * @returns {void}
+   */
+  async open (path) {
+    await this._handleExists(path, `Cannot open file ${path}`)
+    await this._handleIsFile(path, `Cannot open file ${path}`)
+    return this.switchFile(path)
+  }
+
+  /**
+   * Set the content of a specific file
+   * @param {string} path path of the file
+   * @param {string} data content to write on the file
+   * @returns {void}
+   */
+  async writeFile (path, data) {
+    if (await this.exists(path)) {
+      await this._handleIsFile(path, `Cannot write file ${path}`)
+      return await this.setFile(path, data)
+    } else {
+      return await this.setFile(path, data)
+    }
+  }
+
+  /**
+   * Return the content of a specific file
+   * @param {string} path path of the file
+   * @returns {string} content of the file
+   */
+  async readFile (path) {
+    await this._handleExists(path, `Cannot read file ${path}`)
+    await this._handleIsFile(path, `Cannot read file ${path}`)
+    return this.getFile(path)
+  }
+
+  /**
+   * Upsert a file with the content of the source file
+   * @param {string} src path of the source file
+   * @param {string} dest path of the destrination file
+   * @returns {void}
+   */
+  async copyFile (src, dest) {
+    await this._handleExists(src, `Cannot copy from ${src}`)
+    await this._handleIsFile(src, `Cannot copy from ${src}`)
+    await this._handleIsFile(dest, `Cannot paste content into ${dest}`)
+    const content = await this.readFile(src)
+
```

Comment:
how do you readFile or switchFile if it is not exist or is not a file?

Reply:
Should there be provision for that? Because if the file does not exist, there is no point for copy or switchToFile. On the other hand, `writeFile` api handles files that do not exist by creating a new one.